### ISSUE TITLE
Crash when using 'terraform plan' on version 0.66

### ISSUE THREAD

**Issue by catermelon** [Snowflake-Labs/terraform-provider-snowflake#1852]
**Provider Version**

v0.66.0

**Terraform Version**

Terraform v1.4.6

**Describe the bug**

I ran `terraform plan` after configuring snowflake in dev for the first time. It crashed with a stack trace. 

I see it was just released, I'll just downgrade for the meantime. Thanks so much, I appreciate it. 

**Additional context**

> Stack trace from the terraform-provider-snowflake_v0.66.0 plugin:
> 
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x2 addr=0x70 pc=0x100bf2374]
> 
> goroutine 14 [running]:
> github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk.ProfileConfig({0x140006082c0, 0xc})
> 	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/config.go:38 +0xb4
> github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider.DSN({0x0?, 0x14000aa3158?}, {0x0?, 0x14000542880?}, {0x0?, 0x14000aa3178?}, 0xcc?, {0x0?, 0x14000a56820?}, {0x0?, ...}, ...)
> 	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/provider.go:472 +0x1a0
> github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider.ConfigureProvider(0x100f86ec4?)
> 	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/provider.go:360 +0x70c
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0x140002bd380, {0x10154af40, 0x1400069cb70}, 0xd?)
> 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/provider.go:287 +0x16c
> github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0x14000637f08, {0x10154af40?, 0x1400069c360?}, 0x14000a0c390)
> 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:557 +0x314
> github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0x140000f95e0, {0x10154af40?, 0x140002d58c0?}, 0x14000aaec00)
> 	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:556 +0x250
> github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x1014e0800?, 0x140000f95e0}, {0x10154af40, 0x140002d58c0}, 0x14000ab00e0, 0x0)
> 	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
> google.golang.org/grpc.(*Server).processUnaryRPC(0x14000698000, {0x101551a40, 0x14000770000}, 0x14000ab67e0, 0x1400069c300, 0x101e9d058, 0x0)
> 	google.golang.org/grpc@v1.53.0/server.go:1336 +0xb7c
> google.golang.org/grpc.(*Server).handleStream(0x14000698000, {0x101551a40, 0x14000770000}, 0x14000ab67e0, 0x0)
> 	google.golang.org/grpc@v1.53.0/server.go:1704 +0x82c
> google.golang.org/grpc.(*Server).serveStreams.func1.2()
> 	google.golang.org/grpc@v1.53.0/server.go:965 +0x84
> created by google.golang.org/grpc.(*Server).serveStreams.func1
> 	google.golang.org/grpc@v1.53.0/server.go:963 +0x290
> 
> Error: The terraform-provider-snowflake_v0.66.0 plugin crashed!
> 
> This is always indicative of a bug within the plugin. It would be immensely
> helpful if you could report the crash with the plugin's maintainers so that it
> can be fixed. The output above should help diagnose the issue.
>

**Comment by sfc-gh-swinkler** [comment]
Hello @catermelon. I have looked into this issue and have a patch in PR #1853. I will release this ASAP. It appears like your account / username were not set either in provider config, or as an environment variable, or in the ~/.snowflake/config file. Furthermore, there was an error in the ProfileConfig() function which referenced a null pointer if profile wasn't found. I have fixed both issues. Thank you.
